module/VuFind/src/VuFind/OAI/Server.php

```diff
@@ -444,17 +445,9 @@ protected function attachNonDeleted(
         $headerOnly = false,
         $set = ''
     ) {
-        // Get the XML (and display an error if it is unsupported):
-        if ($format === false) {
-            $xml = '';      // no metadata if in header-only mode!
-        } elseif ('oai_vufind_json' === $format && $this->supportsVuFindMetadata()) {
-            $xml = $this->getVuFindMetadata($record);   // special case
-        } else {
-            $xml = $record
-                ->getXML($format, $this->baseHostURL, $this->recordLinkerHelper);
-            if ($xml === false) {
-                return false;
```

Comment:
This edge case of handing the situation where `getXML` returns false (indicating an unsupported format) seems to have been dropped. Is that intentional? I see that there's an `empty($xml)` check below with a comment that seems to address this scenario, but since the refactoring is going to change the return value of this method from `false` to `true` in some situations, I want to be sure it's intentional and that the context has been fully examined. Let me know if it would be helpful for me to do a deeper review on this point!

Reply:
Changed to return instantly, if format is false. Added a default_return key to the array for solving which boolean to return from function.